scim/scim.bal

```diff
@@ -0,0 +1,576 @@
+import ballerina/oauth2;
+import ballerina/http;
+import ballerina/io;
+
+public type SCIMUser record {
```

Comment:
Will need to break down this single file to multiple files. 

1. Check how to separate the data objects to different files in different folders.
2. Include logic related code to different files.